### PR TITLE
cli: allow dump to be used by any user with SELECT

### LINKED ISSUE #7964: cli: cockroach dump only available to root user
When connecting with any user but `root`, you always get this message:

``` shell
Error: pq: user jeldess does not have SELECT privilege on table descriptor
Failed running "dump"
``` 

Only the `root` user has privileges for the `system.descriptor` table, so only the `root` user can run `cockroach dump`. Instead, I would expect any user to be able to dump a table, as long as that user has `SELECT` privileges on the database and table.

**Example:**

``` shell
# Connect as user jseldess:
$ cockroach dump test dumptest --user=jeldess
Error: pq: user jeldess does not have SELECT privilege on table descriptor
Failed running "dump"

# Connect as user root (default):
$ cockroach dump test dumptest
CREATE TABLE dumptest (
	a INT NOT NULL,
	b DECIMAL(10) NULL,
	CONSTRAINT "primary" PRIMARY KEY (a),
	FAMILY "primary" (a, b)
);

INSERT INTO dumptest VALUES
	(1, 1.1),
	(2, 2.2),
	(3, 3.3);
```

### PR DESCRIPTION
**Pull request by madelynnblue**
The previous version used a SELECT to fetch from the system.descriptor
table. This thus required SELECT on that table, which only root has. Instead
of adding some way around this like a new API call, use the existing SHOW
statements to fetch the same data (although not as elegantly).

Fixes #7964

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7974)

### CODE REVIEW

**Comment by tamird** [comment]
Why can't we give everyone SELECT on those tables?

**Comment by tschottdorf** [comment]
Multi-tenancy wouldn't want to let every user see everyone else's schema, I think.

**Comment by tschottdorf** [comment]
Whether that's it or not, a comment somewhere would be good.

**Comment by tamird** [comment]
Right, but then we'd also need to show non-root users to create tables.

**Comment by mjibson** [comment]
Added a comment about multi-tenancy. @tamir either you have a grammar error in your last comment or I don't understand it. Could you rephrase?

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7974#-:-KNIzOcTA5SlEmRuanUW:-216303748)*

**Comment by tamird** [comment]
s/show/allow/

**Comment by tschottdorf** [comment]
LGTM, though I didn't dig into the details so someone more qualified should.

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7974#-:-KNK5_U3DU4p_1lLZXEa:-2082214300)*

**Comment by tschottdorf** [comment]
BTW, add a TODO (or issue) that uses `information_schema` instead once the RFC #7965 is implemented.

**Comment by knz** [comment]
LGTM. +1 on a TODO referring to information_schema.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7974#-:-KNc-LZ1FEX0NvFSjDtw:-1676288658)*

**Comment by jseldess** [comment]
Docs updated with https://github.com/cockroachdb/docs/issues/511
